tests/array.rs

```diff
@@ -96,6 +96,17 @@ fn as_array() {
     })
 }
 
+#[test]
+fn as_raw_array() {
+    pyo3::Python::with_gil(|py| {
+        let not_contiguous = not_contiguous_array(py);
+        let raw_array_view = not_contiguous.as_raw_array();
+        assert_eq!(unsafe { raw_array_view.deref_into_view()[0] }, 1);
```

Comment:
One downside appears to be that ndarray does not provide indexing into raw views, only dereferencing the whole view.

Reply:
Yeah it's not very userfriendly, but I don't think this is rather a downside of raw view, not that of having these APIs itself

Reply:
Opened https://github.com/rust-ndarray/ndarray/pull/1151 to expand the ndarray interface.